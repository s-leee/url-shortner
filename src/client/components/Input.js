import React, { Component } from 'react';
import PropTypes from 'prop-types';

import 'style-loader!./input.less';


class Input extends Component {
	constructor(props) {
		super(props);

		this.state = {
			value: props.value,
		};

		this.onFocus = this.onFocus.bind(this);
		this.onBlur = this.onBlur.bind(this);
		this.onChange = this.onChange.bind(this);
	}

	componentDidMount() {

	}

	onBlur(ev) {
		if (this.props.onBlur) {
			this.props.onBlur(this.state.value);
		}
	}
	onFocus(ev) {
		if (this.props.onFocus) {
			this.props.onFocus(this.state.value);
		}
	}

	onChange(ev) {
		const { value } = ev.target;
		this.setState({ value }, () => {
			if (this.props.onChange) {
				this.props.onChange(value);
			}
		});
	}

	render() {
		const {
			className, placeholder, disabled, readOnly,
		} = this.props;

		return (
			<div className={`comp-input-box ${className}`}>
				<div className={`comp-input-group ${disabled ? 'disabled' : ''} ${readOnly ? 'readonly' : ''}`}>
					<input
						className={'comp-input'}
						onChange={this.onChange}
						value={this.state.value}
						placeholder={placeholder}
						disabled={disabled}
						readOnly={readOnly}
						onFocus={this.onFocus}
						onBlur={this.onBlur}
					/>
				</div>
			</div>
		);
	}
}

Input.propTypes = {
	className: PropTypes.string,
	value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
	placeholder: PropTypes.string,
	disabled: PropTypes.bool,
	readOnly: PropTypes.bool,
	onChange: PropTypes.func.isRequired,
	onFocus: PropTypes.func,
	onBlur: PropTypes.func,
};

Input.defaultProps = {
	className: '',
	value: '',
	placeholder: null,
	disabled: false,
	readOnly: false,
	onFocus: () => {},
	onBlur: () => {},
};

export default Input;
