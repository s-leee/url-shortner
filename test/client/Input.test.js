import React from 'react';
import renderer from 'react-test-renderer';
import Input from '../../src/client/components/Input';

test('Input snapshot', () => {
	const component = renderer.create(
		<Input
			className="shorten-input"
			value="test"
			placeholder="Type in a link to shorten it"
			onChange={(value) => {}}
		/>
	);
	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});
