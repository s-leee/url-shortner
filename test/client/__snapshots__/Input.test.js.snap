// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button onClick 1`] = `
<div
  className="comp-input-box "
>
  <div
    className="comp-input-group  "
  >
    <input
      className="comp-input"
      disabled={false}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder={null}
      readOnly={false}
      value=""
    />
  </div>
</div>
`;

exports[`Input 1`] = `
<div
  className="comp-input-box "
>
  <div
    className="comp-input-group  "
  >
    <input
      className="comp-input"
      disabled={false}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder={null}
      readOnly={false}
      value=""
    />
  </div>
</div>
`;

exports[`Input snapshot 1`] = `
<div
  className="comp-input-box shorten-input"
>
  <div
    className="comp-input-group  "
  >
    <input
      className="comp-input"
      disabled={false}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="Type in a link to shorten it"
      readOnly={false}
      value="test"
    />
  </div>
</div>
`;
